Road/source/features/data_cleaning.py

```diff
@@ -0,0 +1,185 @@
+# import needed libraries
+import pandas as pd 
+import numpy as np 
+import re
+import string 
+
+# Local imports
+# from source.constants import PUNCT_TO_REMOVE
+import sys
+import os
+target = os.path.abspath(__file__)
+while(target.split("\\")[-1]!="Road"):
+    target = os.path.dirname(target)
+sys.path.append(target) 
+
+# Local imports
+from source.constants import PUNCT_TO_REMOVE
+
+# For arabic diacritics 
+from pyarabic.araby import strip_tashkeel, is_tashkeel, is_arabicrange
+
+# For converting emojis
+import emoji
+# pip install emot
+import emot
+
+
+def remove_diacritics(text: str) -> str:
+    """
+    Remove diacritics from Arabic text.
+
+    Args:
+        text (str): The input Arabic text.
+
+    Returns:
+        str: The Arabic text with diacritics removed.
+    """
+    # initialize an empty string to hold the text without diacritics
+    text_without_diacritics = ""
+    for char in text:
+        # check if the character is in the Arabic Unicode range
+        if is_arabicrange(char):
+            # check if the character is a diacritic
+            if is_tashkeel(char):
+                # if it is a diacritic, remove it using the strip_tashkeel function
+                text_without_diacritics += strip_tashkeel(char)
+            else:
+                # if it is not a diacritic, append it to the text without diacritics
+                text_without_diacritics += char
+        else:
+            # if the character is not in the Arabic Unicode range, append it to the text without diacritics
+            text_without_diacritics += char
```

Comment:
No need for the `for` loop, case you can pass a string instead of char to `strip_tashkeel` function.